### DISCUSSION TITLE
Does WAMR support Nuclues OS

### DISCUSSION THREAD

**Discussion by shyamshyre** [bytecodealliance/wasm-micro-runtime#3668] [Q&A]
We have some legacy applications written in C that we’ve been maintaining with minimal updates. Recently, we’ve received customer requests to add new features. The challenge is that these 1990s-era devices have only 1MB of memory, which limits our ability to add new features.

I recently discovered WASM and WAMR and was excited by their core offering of reducing size. I’m exploring WAMR as it seems to be a good fit for our use case and environment. Our current setup is on Nucleus OS, so I’d like to know if WAMR would be helpful in this context.

**Comment by TianlongLiang** [comment]
You can refer to [this document](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/doc/port_wamr.md) to see the necessary steps to port WAMR on a new platform. You may need to add platform API implementation as other platforms did, mostly time and thread-related APIs, you probably don't have to implement all, just those you need

**Reply by shyamshyre** [reply]
Thank you @TianlongLiang  for the response, i'll go over the above article and revert if any further queries.
